Jobsway2goMvc/Controllers/GroupsController.cs

```diff
@@ -35,38 +35,59 @@ public async Task<IActionResult> Details(int? id)
             }
 
             var @group = await _context.Groups
+                .Include(g => g.Posts)
                 .FirstOrDefaultAsync(m => m.Id == id);
             if (@group == null)
             {
                 return NotFound();
             }
+            ViewBag.GroupId = id;
+            return View(group);
+        }
 
-            return View(@group);
+        public IActionResult DetailsPost(int id)
+        {
+            return RedirectToAction("Details", "Post", new { id = id });
+        }
+
+        private ApplicationUser GetApplicationUser(ClaimsPrincipal principal)
+        {
+            var userId = principal.FindFirstValue(ClaimTypes.NameIdentifier);
+
+            var user = _context.Users.FirstOrDefault(u => u.Id == userId);
+
+            return user;
         }
 
-        // GET: Groups/Create
         public IActionResult Create()
         {
             return View();
         }
 
-        // POST: Groups/Create
-        // To protect from overposting attacks, enable the specific properties you want to bind to.
-        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
         [HttpPost]
         [ValidateAntiForgeryToken]
         public async Task<IActionResult> Create([Bind("Id,Name")] Group @group)
         {
+            ModelState.Remove("Posts");
             if (ModelState.IsValid)
             {
+                var userAccessor = _httpContextAccessor.HttpContext.User;
+                var user = GetApplicationUser(userAccessor);
+
+                var groupMembership = new GroupMembership
+                {
+                    Group = @group,
+                    User = user
+                };
+
                 _context.Add(@group);
+                _context.Add(groupMembership);
                 await _context.SaveChangesAsync();
                 return RedirectToAction(nameof(Index));
```

Comment:
`_context.Add(@group);` and then `_context.Add(groupMembership);`? Not sure for this. Create action is only to create groups, and memberships need to have its own action because GroupMembership wasn't meant for the owner/creator of group but for those who join.